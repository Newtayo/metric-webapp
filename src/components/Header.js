import React from 'react';
import { NavLink } from 'react-router-dom';
import './styles/header.css';

const Header = () => (
  <div className="header">
    <NavLink to="/" className="back">
      {' '}
      <i className="fa fa-arrow-left" aria-hidden="true" />
    </NavLink>
    <h1 className="apptitle"> World Population </h1>
    <i className="fa fa-microphone" aria-hidden="true" />
    <i className="fa fa-cog" aria-hidden="true" />
  </div>
);

export default Header;
