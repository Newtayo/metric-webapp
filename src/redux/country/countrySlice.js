import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const url = 'https://restcountries.com/v3.1/all';

const initialState = {

  countryList: [],
  totalCountry: 0,
  isLoading: true,
  error: null,
};

export const getCountries = createAsyncThunk('countryList/getCountries', async (_, thunkAPI) => {
  try {
    const response = await axios.get(url);
    return response.data;
  } catch (error) {
    return thunkAPI.rejectWithValue("Couldn't fetch data");
  }
});

const countrySlice = createSlice({
  name: 'country',
  initialState,

  reducers: {

  },

  extraReducers: (builder) => {
    builder
      .addCase(getCountries.pending, (state) => ({
        ...state,
        isLoading: true,
      }))
      .addCase(getCountries.fulfilled, (state, action) => {
        const country = action.payload;
        const countryArray = [];
        let counter = 0;
        country.map((details) => {
          counter += details.population;
          countryArray.push({
            id: details.name.official,
            countryName: details.name.common,
            countryFlag: details.flags.png,
            population: details.population,
            capital: details.capital,
            continents: details.continents,
            coatofarm: details.coatOfArms,
            area: details.area,

          });
          return countryArray;
        });

        return {
          ...state,
          countryList: countryArray,
          totalCountry: counter,
          isLoading: false,
        };
      })
      .addCase(getCountries.rejected, (state, action) => ({
        ...state,
        isLoading: false,
        error: action.payload,
      }));
  },
});

export const { totalpopulation } = countrySlice.actions;

export default countrySlice.reducer;
