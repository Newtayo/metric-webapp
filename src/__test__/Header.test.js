import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Store from '../redux/store';
import Header from '../components/Header';
import MockAxios, { mockResponse } from '../__mocks__/axios';
import { getCountries } from '../redux/country/countrySlice';

afterEach(cleanup);
describe('tests for components', () => {
  test('Snapshot test for Header', () => {
    const homepage = render(
      <Provider store={Store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );
    expect(homepage).toMatchSnapshot();
  });

  test('Snapshot test for Header', async () => {
    jest.mock('axios');

    MockAxios.get.mockResolvedValue(mockResponse);

    render(
      <Provider store={Store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );

    const { type } = await Store.dispatch(getCountries());
    expect(type).toEqual('countryList/getCountries/fulfilled');
    expect(Store.getState().country.countryList[0].countryName).toContain('Barbados');
    expect(Store.getState().country.totalCountry).toEqual(287371);
  });
});
