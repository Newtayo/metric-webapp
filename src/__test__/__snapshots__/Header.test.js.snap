// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for components Snapshot test for Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <a
          aria-current="page"
          class="back active"
          href="/"
        >
           
          <i
            aria-hidden="true"
            class="fa fa-arrow-left"
          />
        </a>
        <h1
          class="apptitle"
        >
           World Population 
        </h1>
        <i
          aria-hidden="true"
          class="fa fa-microphone"
        />
        <i
          aria-hidden="true"
          class="fa fa-cog"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <a
        aria-current="page"
        class="back active"
        href="/"
      >
         
        <i
          aria-hidden="true"
          class="fa fa-arrow-left"
        />
      </a>
      <h1
        class="apptitle"
      >
         World Population 
      </h1>
      <i
        aria-hidden="true"
        class="fa fa-microphone"
      />
      <i
        aria-hidden="true"
        class="fa fa-cog"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
